src/Core/Config/Config.php

```diff
@@ -119,4 +118,23 @@ public static function forClass($class)
     {
         return new Config_ForClass($class);
     }
+
+    /**
+     * Perform the given operation in an isolated config state.
+     * On return, the config state will be restored, so any modifications are temporary.
+     *
+     * @param callable $callback Callback to run. Will be passed the nested config state as a parameter
+     * @return mixed Result of callback
+     */
+    public static function withConfig($callback)
+    {
+        static::nest();
+        $config = static::modify();
+
+        try {
+            return $callback($config);
```

Comment:
Could this be using `return call_user_func($callback, $config)` to support more ways of providing a valid callback, e.g. an array of class & method pair?